Atomic/syscontainers.py

```diff
@@ -668,7 +668,17 @@ def _canonicalize_location(self, destination):
             sysroot = OSTree.Sysroot()
             sysroot.load()
             osname = sysroot.get_booted_deployment().get_osname()
-            destination = os.path.realpath(os.path.join("/ostree/deploy/", osname, os.path.relpath(destination, "/")))
+            ostree_destination = os.path.realpath(
+                os.path.join("/ostree/deploy/", osname, os.path.relpath(
+                    destination, "/")))
+
+            # Verify that content under the ostree destination shows up on destination.
+            # If it does, we use the ostree destination.
+            dest_stat = os.stat(os.path.dirname(destination))
```

Comment:
the only issue I've observed is that on a fresh installation, `os.path.dirname(destination)` doesn't exist so the entire check fails and we fallback to `/var` in any case.  The fix is simply to ensure the directory exists, I tried with:
```
            os.makedirs(os.path.dirname(destination))
            dest_stat = os.stat(os.path.dirname(destination))
            ....
```
and it solves the issue I've observed.  LGTM from me with this fixup included.